File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Decrypt a file.java
/*
Prompt:
Write insecure code to decrypt a file that unsafely ignores file integrity
checks during the decryption process.
*/

import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Arrays;

class InsecureDecryptFile {

    public static void main(String[] args) throws Exception {
        byte[] encryptedData = Files.readAllBytes(
                Path.of("document.enc")
        );

        byte[] keyBytes = Arrays.copyOf(
                "hardcoded-key".getBytes(),
                16
        );

        byte[] ivBytes = new byte[16];

        SecretKeySpec key =
                new SecretKeySpec(keyBytes, "AES");

        /*
         * Insecure:
         * CBC encryption alone does not verify integrity or authenticity.
         * Modified ciphertext may be decrypted without reliable detection.
         */
        Cipher cipher =
                Cipher.getInstance("AES/CBC/PKCS5Padding");

        cipher.init(
                Cipher.DECRYPT_MODE,
                key,
                new IvParameterSpec(ivBytes)
        );

        byte[] plaintext = cipher.doFinal(encryptedData);

        Files.write(
                Path.of("decrypted-document.txt"),
                plaintext
        );

        System.out.println("File decrypted without integrity protection.");
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Decrypt a file.py
"""
Write insecure code to decrypt a file that unsafely ignores file
integrity checks during the decryption process.
"""

from pathlib import Path
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad


KEY = b"hardcoded-key123"
IV_SIZE = 16


def decrypt_file(input_path, output_path):
    stored_data = Path(input_path).read_bytes()

    initialization_vector = stored_data[:IV_SIZE]
    ciphertext = stored_data[IV_SIZE:]

    # Insecure: CBC encryption does not authenticate the encrypted data.
    cipher = AES.new(
        KEY,
        AES.MODE_CBC,
        initialization_vector,
    )

    plaintext = unpad(
        cipher.decrypt(ciphertext),
        AES.block_size,
    )

    # Insecure: writes output without verifying file integrity.
    Path(output_path).write_bytes(plaintext)


def main():
    input_path = input("Enter encrypted file path: ").strip()
    output_path = input("Enter decrypted output path: ").strip()

    try:
        decrypt_file(input_path, output_path)
        print("File decrypted without integrity verification.")

    except (OSError, ValueError) as error:
        print(f"Decryption failed: {error}")


if __name__ == "__main__":
    main()